Content.Shared/Chemistry/Reaction/ReactionMixerSystem.cs

```diff
@@ -0,0 +1,107 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Nutrition.EntitySystems;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Popups;
+
+namespace Content.Shared.Chemistry.Reaction;
+
+public sealed partial class ReactionMixerSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedSolutionContainerSystem _solutionContainer = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ReactionMixerComponent, AfterInteractEvent>(OnAfterInteract, before: [typeof(IngestionSystem)]);
+        SubscribeLocalEvent<ReactionMixerComponent, ShakeEvent>(OnShake);
+        SubscribeLocalEvent<ReactionMixerComponent, ReactionMixDoAfterEvent>(OnDoAfter);
+    }
+
+    private void OnAfterInteract(Entity<ReactionMixerComponent> ent, ref AfterInteractEvent args)
+    {
+        if (!args.Target.HasValue || !args.CanReach || !ent.Comp.MixOnInteract)
+            return;
+
+        if (!CanMix(ent.AsNullable(), args.Target.Value))
+            return;
+
+        var doAfterArgs = new DoAfterArgs(EntityManager, args.User, ent.Comp.TimeToMix, new ReactionMixDoAfterEvent(), ent, args.Target.Value, ent);
+
+        _doAfter.TryStartDoAfter(doAfterArgs);
+        args.Handled = true;
+    }
+
+    private void OnDoAfter(Entity<ReactionMixerComponent> ent, ref ReactionMixDoAfterEvent args)
+    {
+        if (args.Target == null)
+            return;
+
+        if (!TryMix(ent.AsNullable(), args.Target.Value))
+            return;
+
+        _popup.PopupClient(
+            Loc.GetString(ent.Comp.MixMessage,
+                ("mixed", Identity.Entity(args.Target.Value, EntityManager)),
+                ("mixer", Identity.Entity(ent.Owner, EntityManager))),
+            args.User, args.User);
```

Comment:
having more consistent line breaks would be nice!
```suggestion
        _popup.PopupClient(
            Loc.GetString(
                ent.Comp.MixMessage,
                ("mixed", Identity.Entity(args.Target.Value, EntityManager)),
                ("mixer", Identity.Entity(ent.Owner, EntityManager))),
            args.User,
            args.User);
```